gcloud/monitoring/test_client.py

```diff
@@ -157,6 +157,79 @@ def test_metric_descriptor_factory(self):
         self.assertEqual(descriptor.description, DESCRIPTION)
         self.assertEqual(descriptor.display_name, '')
 
+    def test_metric_factory(self):
+        TYPE = 'custom.googleapis.com/my_metric'
+        LABELS = {
+            'instance_name': 'my-instance'
+        }
+
+        client = self._makeOne(project=PROJECT, credentials=_Credentials())
+        client.connection = _Connection()   # For safety's sake.
+        metric = client.metric(TYPE, LABELS)
+        self.assertEqual(metric.type, TYPE)
+        self.assertEqual(metric.labels, LABELS)
+
+    def test_resource_factory(self):
+        TYPE = 'https://cloud.google.com/monitoring/api/resources'
+        LABELS = {
+            'project_id': 'my-project-id',
+            'instance_id': 'my-instance-id',
+            'zone': 'us-central1-f'
+        }
+
+        client = self._makeOne(project=PROJECT, credentials=_Credentials())
+        client.connection = _Connection()   # For safety's sake.
+        resource = client.resource(TYPE, LABELS)
+        self.assertEqual(resource.type, TYPE)
+        self.assertEqual(resource.labels, LABELS)
+
+    def test_timeseries_factory(self):
+        import datetime
+        from gcloud._testing import _Monkey
+        import gcloud.monitoring.client
+        METRIC_TYPE = 'custom.googleapis.com/my_metric'
+        METRIC_LABELS = {
+            'instance_name': 'my-instance'
+        }
+
+        RESOURCE_TYPE = 'https://cloud.google.com/monitoring/api/resources'
+        RESOURCE_LABELS = {
+            'project_id': 'my-project-id',
+            'instance_id': 'my-instance-id',
+            'zone': 'us-central1-f'
+        }
+
+        VALUE = 42
+
```

Comment:
There are three cases in this test: 1) A gauge with an explicit end time, 2) a cumulative metric, and 3) a gauge with an implicit end time. We should take the opportunity to teach how cumulative metrics work. The start time is the "reset time", which is typically going to be the time at which this instance of the application started up.

I would start by defining constants here named `RESET_TIME`, `TIME1`, `TIME2`, and `TIME3`. I would just construct four arbitrary `datetime` objects. That seems better form than using `utcnow()`. Then see my further comments below.